android/src/main/java/com/rivereactnative/RiveReactNativeView.kt

```diff
@@ -90,6 +105,25 @@ class RiveReactNativeView(private val context: ThemedReactContext) : FrameLayout
     addView(riveAnimationView)
   }
 
+  override fun onAttachedToWindow() {
+    // RiveReactNativeView may attach multiple times for any reason, so using
+    // a flag here isAttached to determine when it was initially attached to
+    // only addView once
+    if (childCount == 0 && isAttached == false) {
+      addView(riveAnimationView)
+      update()
+      isAttached = true
+    }  
+    super.onAttachedToWindow()
+  }
+
+
+  override fun onDetachedFromWindow() {
+    // Starting in Rive Android 4.0+, we need to coordinate removing this view with
+    // the underlying RiveAnimationView when resources are cleaned up
+    removeView(riveAnimationView)
+    super.onDetachedFromWindow();
```

Comment:
Setting to `false` would cause some kind of cyclical attach/detach (since after detaching, it would go to attach and then add the view again) and specifically when officially detaching this view, this would cause the cpp memory reference error when everything has been cleaned up on Android side of things.

I think perhaps a better name for this would be `hasMounted` to be more distinctive from the detach/attach handlers here?